### ISSUE TITLE
An undefined bug?

### ISSUE THREAD

**Issue by iaxax** [x-motemen/gore#189]
![undefined](https://user-images.githubusercontent.com/14243516/82651179-63903a00-9c4e-11ea-83cf-50f689789ee5.png)

**Comment by itchyny** [comment]
This is surely a surprising behavior. Currently gore tries to 1. parse the input as an expression, 2. parse as a declaration in the main function 3. declaration outside the main function. The struct declaration in the main function is allowed in Go so the input generates the following code. But maybe we can fix this without breaking other behaviors.

```go
package main

func main()	{ type A struct{ v int } }
func foo(a A)	{ a.v = 2 }
```

**Comment by xiantang** [comment]
https://github.com/x-motemen/gore/issues/34#issuecomment-991494511
struct method also cannot define...

**Comment by xiantang** [comment]
can we define function inside main function?

**Comment by xiantang** [comment]
can we define struct out side the main function. then struct method and function reference costom struct can use correct.
will it be breaking change?

**Comment by xiantang** [comment]
![image](https://user-images.githubusercontent.com/34479567/145704868-5796407c-dc04-48ea-b643-460865f0e52f.png)
I have support this in my side

**Comment by xiantang** [comment]
can try my version https://github.com/xiantang/gore.. 
I have supported method and pointer function for struct, but dont know whether it's a breaking change..
if you want use this feature can clone my fork than `make install`


@itchyny

**Comment by itchyny** [comment]
This has been resolved in v0.5.5.
